test/test_id_model.cpp

```diff
@@ -849,9 +974,66 @@ TEST_F(IdModelTest, LoopPromotion5) {
   // Check Step 2 results
   checkStep2Results(
       &fusion,
-      iel_graph,
+      tester.iel_graph,
       tester.idGraph(IdMappingMode::EXACT),
-      iel_promotion_map);
+      tester.s2_iel_promotion_map);
+
+  // Check Step 3 results. See the design doc for the expected results
+  std::vector<std::pair<std::unordered_set<Val*>, IterDomain*>>
+      s3_reference_map = {
+          // 7, 10, 11, 25, 14, 15, 18 -> 18
+          {std::unordered_set<Val*>{
+               tv2->getRootDomain().at(0),
+               tv3->getRootDomain().at(0),
+               tv3->getRootDomain().at(1),
+               getParentId(tv3->axis(0), 4),
+               tv4->getRootDomain().at(0),
+               tv4->getRootDomain().at(1),
+               getParentId(tv4->axis(0), 4)},
+           getParentId(tv4->axis(0), 4)},
+          // 8, 12, 16 -> 16
+          {std::unordered_set<Val*>{
+               tv2->getRootDomain().at(1),
+               tv3->getRootDomain().at(2),
+               tv4->getRootDomain().at(2)},
+           tv4->getRootDomain().at(2)},
+          // 9, 13, 17 -> 17
+          {std::unordered_set<Val*>{
+               tv2->getRootDomain().at(2),
+               tv3->getRootDomain().at(3),
+               tv4->getRootDomain().at(3)},
+           tv4->getRootDomain().at(3)},
+          // 32, 26, 19 -> 19
+          {std::unordered_set<Val*>{
+               getParentId(tv2->axis(0), 3),
+               getParentId(tv3->axis(0), 3),
+               getParentId(tv4->axis(0), 3)},
+           getParentId(tv4->axis(0), 3)},
```

Comment:
nit: would it be more readable if we use:
```suggestion
               getChildId(tv2->getRootDomain().at(0), 1),
               getChildId(tv3->getRootDomain().at(2), 1),
               getChildId(tv3->getRootDomain().at(2), 1)},
           getChildId(tv3->getRootDomain().at(2), 1)},
```